controller/src/main/java/io/pravega/controller/store/index/ZKHostIndex.java

```diff
@@ -160,6 +160,24 @@ public ZKHostIndex(CuratorFramework client, String hostRoot, Executor executor)
         return result;
     }
 
+    CompletableFuture<Void> sync(final String path) {
```

Comment:
I understand that it is difficult to recreate the issue you are trying to solve here in a unit test, but I suggest that we at least unit test the new `sync` method implementations so that we we don't drop code coverage.